File: programmers/02.stack,queue/03. 기능개발/JS/01.js
function solution(progresses, speeds) {
    const deployDay = progresses.map((func, idx) =>
        Math.ceil((100 - func) / speeds[idx])
    );
    console.log(deployDay);

    let answer = [];
    let max = deployDay[0];
    let cnt = 1;
    for (let i = 1; i < deployDay.length; i++) {
        if (deployDay[i] <= max) cnt++;
        else {
            max = deployDay[i];
            answer.push(cnt);
            cnt = 1;
        }
    }
    answer.push(cnt);
    return answer;
}

// Test
console.log(solution([93, 30, 55], [1, 30, 5])); // [ 7, 3, 9 ] => [2, 1]
console.log(solution([95, 90, 99, 99, 80, 99], [1, 1, 1, 1, 1, 1])); // [ 5, 10, 1, 1, 20, 1 ] => [1,3,2]

File: programmers/01.알고리즘/02.stack,queue/01.같은 숫자는 싫어 /python/01.py
def solution(arr):
    answer = []
    arr.append(10)
    for (i, e) in enumerate(arr[:-1]):
        if e != arr[i+1]:
            answer.append(e)
    return answer

# test
print(solution([1,1,3,3,0,1,1]))
print(solution([4,4,4,3,3]))